File: backend/ShiftMate.API/Models/Employee.cs
namespace ShiftMate.API.Models;

public class Employee
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public int DepartmentId { get; set; }

    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string? Phone { get; set; }

    public DateTime HireDate { get; set; }

    public bool IsActive { get; set; }

    public Company Company { get; set; } = null!;

    public Department Department { get; set; } = null!;
}

File: frontend/src/types/employee.ts
export interface Employee {
    id: number
    departmentId: number
    companyId: number
    firstName: string
    lastName: string
    email: string
    phone: string | null
    hireDate: string
    isActive: boolean
}

export interface CreateEmployeeRequest {
    firstName: string
    lastName: string
    email: string
    phone: string 
    departmentId: number
    hireDate: string
}

export interface UpdateEmployeeRequest {
    firstName: string
    lastName: string
    email: string
    phone: string
    departmentId: number
    hireDate: string
    isActive: boolean
}